src/fsharp/ConstraintSolver.fs

```diff
@@ -1914,67 +1914,73 @@ and CanMemberSigsMatchUpToCheck
-                let reflArgInfo = calledArg.ReflArgInfo // propgate the reflected-arg info to each param array argument
-                calledMeth.ParamArrayCallerArgs |> OptionD (IterateD (fun callerArg -> subsumeArg (CalledArg((0, 0), false, NotOptional, NoCallerInfo, false, false, None, reflArgInfo, paramArrayElemTy)) callerArg))
-            else
-                CompleteD)
-        
-        ) ++ (fun () -> 
-        (calledMeth.ArgSets |> IterateD (fun argSet -> 
-            argSet.AssignedNamedArgs |> IterateD (fun arg -> subsumeArg arg.CalledArg arg.CallerArg)))  ++ (fun () -> 
-        (assignedItemSetters |> IterateD (fun (AssignedItemSetter(_, item, caller)) -> 
-            let name, calledArgTy = 
-                match item with
-                | AssignedPropSetter(_, pminfo, pminst) -> 
-                    let calledArgTy = List.head (List.head (pminfo.GetParamTypes(amap, m, pminst)))
-                    pminfo.LogicalName, calledArgTy
-
-                | AssignedILFieldSetter(finfo) ->
-                    (* Get or set instance IL field *)
-                    let calledArgTy = finfo.FieldType(amap, m)
-                    finfo.FieldName, calledArgTy
+                    if calledObjArgTys.Length <> callerObjArgTys.Length then 
+                        if (calledObjArgTys.Length <> 0) then
+                            ErrorD(Error (FSComp.SR.csMemberIsNotStatic(minfo.LogicalName), m))
+                        else
+                            ErrorD(Error (FSComp.SR.csMemberIsNotInstance(minfo.LogicalName), m))
+                    else
+                        Iterate2D subsumeTypes calledObjArgTys callerObjArgTys
+            do! calledMeth.ArgSets
+                |> IterateD (fun argSet ->
+                    if argSet.UnnamedCalledArgs.Length <> argSet.UnnamedCallerArgs.Length then ErrorD(Error(FSComp.SR.csArgumentLengthMismatch(), m)) else
+                    Iterate2D subsumeArg argSet.UnnamedCalledArgs argSet.UnnamedCallerArgs)
+            do! calledMeth.ParamArrayCalledArgOpt
+                |> OptionD 
+                    (fun calledArg ->
+                        if isArray1DTy g calledArg.CalledArgumentType then 
+                            let paramArrayElemTy = destArrayTy g calledArg.CalledArgumentType
+                            let reflArgInfo = calledArg.ReflArgInfo // propgate the reflected-arg info to each param array argument
+                            calledMeth.ParamArrayCallerArgs |> OptionD (IterateD (fun callerArg -> subsumeArg (CalledArg((0, 0), false, NotOptional, NoCallerInfo, false, false, None, reflArgInfo, paramArrayElemTy)) callerArg))
+                        else
+                            CompleteD)
+            do! calledMeth.ArgSets 
+                |> IterateD (fun argSet -> 
+                    argSet.AssignedNamedArgs |> IterateD (fun arg -> subsumeArg arg.CalledArg arg.CallerArg))
+            do! assignedItemSetters 
```

Comment:
Can you use a `for` loop here, since `trackErrors` has a `For` method?

Reply:
we need to implement more methods so the for do is usable.